newIDE/app/src/UI/Carousel.js

```diff
@@ -0,0 +1,329 @@
+              <Text noMargin style={styles.itemTitle}>
+                {item.title}
+              </Text>
+            </div>
+          ) : (
+            <Skeleton
+              variant="rect"
+              height={titleHeight}
+              width={(cellWidth / 3) * (1 + 2 * Math.random())}
+            />
+          )}
+        </>
+      );
+    },
+    [cellWidth, titleHeight]
+  );
+
+  const computeScroll = React.useCallback(
+    (
+      direction: 'left' | 'right',
+      scrollViewElement: HTMLUListElement
+    ): number => {
+      const visibleThumbnailsCount = Math.floor(
+        scrollViewElement.offsetWidth / widthUnit
+      );
+      const scale = visibleThumbnailsCount * widthUnit;
+
+      const currentScroll = scrollViewElement.scrollLeft;
+      const currentFirstVisibleItemIndex = currentScroll / widthUnit;
+
+      if (
+        direction === 'right' &&
+        currentFirstVisibleItemIndex >
+          itemsToDisplay.length - visibleThumbnailsCount - 1
+      )
+        return 0;
+      return roundScroll(
+        scrollViewElement.scrollLeft + scale * (direction === 'left' ? -1 : 1)
+      );
+    },
+    [widthUnit, itemsToDisplay]
+  );
+
+  const roundScroll = React.useCallback(
+    (value: number): number => {
+      return Math.round(value / widthUnit) * widthUnit;
+    },
+    [widthUnit]
+  );
+
+  const onClickArrow = (direction: 'left' | 'right') =>
+    React.useCallback(
+      (): void => {
+        const scrollViewElement = scrollView.current;
+        if (!scrollViewElement) return;
+        const newScrollPosition = computeScroll(direction, scrollViewElement);
+
+        scrollViewElement.scrollTo({
+          left: newScrollPosition,
+          behavior: 'smooth',
+        });
+      },
+      [computeScroll]
+    );
+
+  const handleScroll = React.useCallback(() => {
+    const scrollViewElement = scrollView.current;
+    if (!scrollViewElement) return;
+
+    if (!shouldDisplayLeftArrow)
+      setShouldDisplayLeftArrow(scrollViewElement.scrollLeft !== 0);
+  });
+  const handleScrollEnd = React.useCallback(
+    () => {
+      const scrollViewElement = scrollView.current;
+      if (!scrollViewElement) return;
+
+      scrollViewElement.scrollTo({
+        left: roundScroll(scrollViewElement.scrollLeft),
+        behavior: 'smooth',
+      });
+    },
+    [roundScroll]
+  );
+
+  React.useEffect(() => {
+    const scrollViewElement = scrollView.current;
+    if (!scrollViewElement) return;
+
+    scrollViewElement.addEventListener('scroll', handleScroll);
+    scrollViewElement.addEventListener('touchend', handleScrollEnd);
+    scrollViewElement.addEventListener('touchleave', handleScrollEnd);
+  }, []);
+
+  return (
+    <Line noMargin expand>
+      <div
+        style={{ ...styles.arrowContainer, width: arrowWidth }}
+        onClick={onClickArrow('left')}
+      >
+        {!!items && shouldDisplayLeftArrow && <ArrowBackIos />}
```

Comment:
should all the checks on `items` be more `items && items.length` ? As this is an array, [] will be truthy

Reply:
Good catch, I'm on it